rendercv/data_model.py

```diff
@@ -1048,13 +1048,99 @@ class Connection(BaseModel):
         "GitHub",
         "Instagram",
         "Orcid",
+        "Mastodon",
         "phone",
         "email",
         "website",
         "location",
     ]
     value: str
 
+    @staticmethod
+    def is_valid_hostname(hostname: str) -> bool:
+        """Is hostname a valid hostname by RFCs 952 and 1123"""
+
+        # slightly modified from
+        # https://stackoverflow.com/a/33214423/1304076
+        if hostname[-1] == ".":
+            # strip exactly one dot from the right, if present
+            hostname = hostname[:-1]
+        if len(hostname) > 253:
+            return False
+
+        labels = hostname.split(".")
+
+        # the last label must be not all-numeric
+        if re.match(r"[0-9]+$", labels[-1]):
+            return False
+
+        # labels cannot begin with a hyphen
+        # labels must have at least character
+        # labels may not have more than 63 characters
+        allowed = re.compile(r"(?!-)[a-z0-9-]{1,63}(?<!-)$", re.IGNORECASE)
+        return all(allowed.match(label) for label in labels)
+
```

Comment:
If we need to verify Mastodon usernames, we should also verify other networks. Is it necessary to validate usernames? Moreover, if we want to do a validation, we should use the @field_validator decorator of Pydantic. This will help exception handling since we will know it's a validation error.

Reply:
There are two separate suggestions. The reason that Mastodon needs special validation as opposed to, say, LinkedIn is that we know that know that `www.linkedin.com` ia  syntactically valid hostname. We aren't going to create a malicious PDF by using it.

There may be value in syntactically validating other usernames. But I do think it would be a mistake to actually perform any network calls while running RenderCV. So the validation is only syntactic.

I very much suspect you are correct about `@field_validator`, but the be perfectly honest, I don't fully grok what Pydantic is doing. However, I am happy to learn. So I will work on this.